### PR TITLE
[PWX-36732] vendor update kvdb

### PR DESCRIPTION
**Pull request by Glitchfix**
**What this PR does / why we need it**:
udpate go packages majorly etcd and its dependencies
It will also be needed to update dependent packages like secrets, sched-ops, etc. which would help us resolve conflicting versions of certain packages like gRPC and will help us be at the latest versions of all the other packages.

**Which issue(s) this PR fixes** (optional)
Closes #[PWX-36732](https://purestorage.atlassian.net/browse/PWX-36732)

[PWX-36732]: https://purestorage.atlassian.net/browse/PWX-36732?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: etcd/v2/kv_etcd.go [hunk 1]
```diff
@@ -1,1003 +0,0 @@
-package etcdv2
```
Comment (dahuang-purestorage) [inline etcd/v2/kv_etcd.go:1]: why was this deleted? just curious

**Comment by dahuang-purestorage** [comment]
Can you put more description on why we are doing this change? For example, in porx, there's conflict in x,y,z.. Changes in the PR would resolve x,y,z because ....
Many thanks

File: go.mod [hunk 2]
```diff
@@ -1,17 +1,29 @@
 module github.com/portworx/kvdb
 
-go 1.17
+go 1.20
+
+require (
```
Comment (dahuang-purestorage) [inline go.mod:5]: why is this section apart from the require section below? 
If you run `go mod tidy` or `go mod vendor`, they should organize it for you.

File: etcd/v2/kv_etcd.go (hunk 1 above)
Comment (Glitchfix) [reply etcd/v2/kv_etcd.go:1]: I asked a question in the channel, and @adityadani confirmed that it is ok to deprecate the etcd2 reference in our code.

File: go.mod (hunk 2 above)
Comment (Glitchfix) [reply go.mod]: tried both but I think it doesn't organise automatically, however I think this wouldn't be a problem
Also, I have updated it by hand now

File: go.mod [hunk 3]
```diff
@@ -1,17 +1,29 @@
...
+	go.etcd.io/etcd v0.5.0-alpha.5.0.20200910180754-dd1b699fc489
+	google.golang.org/genproto v0.0.0-20201210142538-e3217bee35cc // indirect
+	google.golang.org/grpc v1.34.0
+)
+
+replace (
+	github.com/coreos/etcd => github.com/coreos/etcd v3.3.13+incompatible
+	go.etcd.io/bbolt => go.etcd.io/bbolt v1.3.5
+	go.etcd.io/etcd => go.etcd.io/etcd v0.5.0-alpha.5.0.20200910180754-dd1b699fc489 // ae9734ed278b is the SHA for git tag v3.4.13
```
[the remark this answers is not in the document]
Comment (Glitchfix) [reply go.mod]: updated it to v3.3.27+incompatible

**Review by dahuang-purestorage — APPROVED** [review]
LGTM
